internal/host/validations_test.go

```diff
@@ -374,6 +378,132 @@ var _ = Describe("Validations test", func() {
 			}
 		}
 
+		Describe("Wildcard connectivity check is performed", func() {
+			successMessage := "DNS wildcard check was successful"
+			successMessageDay2 := "DNS wildcard check is not required for day2"
+			failureMessage := "DNS wildcard configuration was detected for domain *.test-cluster.example.com The installation will not be able to complete while the entry exists. Please remove it to proceed."
+			errorMessage := "Error while parsing DNS resolution response"
+			pendingMessage := "DNS wildcard check cannot be performed yet because the host has not yet performed DNS resolution"
+
+			for _, dnsTestCase := range []struct {
```

Comment:
I'm really not a fan of these kinds of tests.

I think it would be more clear to have each example separate. Having the majority of the actual test wrapped in conditionals makes it pretty hard to see what's going on.
There doesn't seem to be too much overlap between the cases as most of the setup is already extracted into helpers.

Reply:
Fair. I've improved the naming / flow significantly, let me know if you still think it's hard to see what's going on